### PR TITLE
Added a test case for postLoad on fetch-joined entities - fetch joined entities should have lifecycle events fired as well

### PR DESCRIPTION
**Pull request by stoccc**
see https://github.com/doctrine/doctrine2/issues/6568

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Functional/LifecycleCallbackTest.php [hunk 1]
```diff
@@ -268,6 +268,47 @@ public function testCascadedEntitiesNotLoadedInPostLoadDuringIterationWithSimple
...
+
+        $dql = <<<'DQL'
+SELECT
+    entA, entB
+FROM
+    Doctrine\Tests\ORM\Functional\LifecycleCallbackCascader AS entA
+LEFT JOIN
+    entA.entities AS entB
+WHERE
+    entA.id = %s
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Functional/LifecycleCallbackTest.php]: Please use a named parameter instead of `sprintf()`

**Comment by Ocramius** [comment]
Test looks good - let's see if it fails on CI. If that's the case, we got some hydrator hell to fix :-\

**Comment by stoccc** [comment]
@Ocramius 
think you should remove Bug and Failing Test labels.
I closed the bug, you gained a test case if it can help.

**Comment by Ocramius** [comment]
Marketing as `Improvement` then

**Pull request merged**